bundles/org.openhab.binding.mideaac/src/main/java/org/openhab/binding/mideaac/internal/discovery/MideaACDiscoveryService.java

```diff
@@ -0,0 +1,370 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.mideaac.internal.discovery;
+
+import static org.openhab.binding.mideaac.internal.MideaACBindingConstants.*;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.SocketException;
+import java.net.SocketTimeoutException;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.TreeMap;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.mideaac.internal.Utils;
+import org.openhab.binding.mideaac.internal.dto.CloudProviderDTO;
+import org.openhab.binding.mideaac.internal.handler.CommandBase;
+import org.openhab.binding.mideaac.internal.security.Security;
+import org.openhab.core.config.discovery.AbstractDiscoveryService;
+import org.openhab.core.config.discovery.DiscoveryResult;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.config.discovery.DiscoveryService;
+import org.openhab.core.thing.ThingUID;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link MideaACDiscoveryService} service for Midea AC.
+ *
+ * @author Jacek Dobrowolski - Initial contribution
+ * @author Bob Eckhoff - OH naming conventions
+ */
+@NonNullByDefault
+@Component(service = DiscoveryService.class, configurationPid = "discovery.mideaac")
+public class MideaACDiscoveryService extends AbstractDiscoveryService {
+
+    private static int discoveryTimeoutSeconds = 5;
+    private final int receiveJobTimeout = 20000;
+    private final int udpPacketTimeout = receiveJobTimeout - 50;
+    private final String mideaacNamePrefix = "MideaAC";
+
+    private final Logger logger = LoggerFactory.getLogger(MideaACDiscoveryService.class);
+
+    ///// Network
+    private byte[] buffer = new byte[512];
+    @Nullable
+    private DatagramSocket discoverSocket;
+
+    @Nullable
+    DiscoveryHandler discoveryHandler;
+
+    private Security security;
+
+    private Map<String, Object> initializeProperties() {
+        final Map<String, Object> properties = new TreeMap<>();
+        properties.put(CONFIG_IP_ADDRESS, "");
+        properties.put(CONFIG_IP_PORT, "0");
+        properties.put(CONFIG_DEVICEID, "");
+        properties.put(CONFIG_POLLING_TIME, 60);
+        properties.put(CONFIG_CONNECTING_TIMEOUT, 4);
+        properties.put(CONFIG_PROMPT_TONE, false);
```

Comment:
Please don't duplicate the configuration parameters to the Thing properties.
```suggestion
```

Reply:
This was an earlier attempt to eliminate the null warnings.  I ended up doing something AI inspired similar to your suggestion below that you can review once I commit again.  I'll take these out.

Reply:
Might be forgotten, they are still there.

Reply:
No problem with the properties that are configurations with defaults. I guess what this points out that I don't know is how the other properties that are discovered here get added/transferred to the configuration.  I thought it was through the `DiscoveryResultBuilder` that uses the items in this method to populate the configuration?

Reply:
You are right.  These remaining configuration options need to be added as property for the discovery result to work.